contrib/cirrus/lib.sh

```diff
@@ -298,9 +298,17 @@ setup_rootless() {
     # Guarantee independence from specific values
     rootless_uid=$[RANDOM+1000]
     rootless_gid=$[RANDOM+1000]
-    msg "creating $rootless_uid:$rootless_gid $ROOTLESS_USER user"
+    rootless_supplemental_gid1=$[RANDOM+1000]
+    rootless_supplemental_gid2=$[RANDOM+1000]
+    rootless_supplemental_gid3=$[RANDOM+1000]
```

Comment:
Isn't this just introducing a setup flake? If we randomly get the same gid again it would not add extra gids, I just tested useradd and it doesn't error in this case.

However the groupadd calls will fails since the group already exists. It is not likely to happen but I think you must make sure uses a different uid range for each group to avoid the conflict.

And the fact that this does not trigger https://github.com/containers/buildah/pull/6226 is concerning no? It means we are not testing what we think we are testing with this. The fact that the tesst does not log a skip message makes it impossible to tell if the "--group-add keep-groups" check was run or not because it is inside a if that does not use skip

Reply:
Hmm, not catching duplicates would introduce a flake, that's worth fixing.  Maybe logging what's going on when the tests aren't triggered will shed a clue.